### PR TITLE
Fix: UrlInput autoFocus Issue With Custom Block

### PR DESCRIPTION
**Pull request by yahilmadakiya**
Fix: #6196 

Added option to set autofocus true/false. Default can be true.

### CODE REVIEW

**Review by danielbachhuber — CHANGES_REQUESTED** [review]
Thanks @yahilmadakiya ! Can you include unit tests for this change?

**Comment by yahilmadakiya** [comment]
Sure @danielbachhuber, I'll try to add unit test.

**Comment by yahilmadakiya** [comment]
@danielbachhuber I have tried a unit test for the changes.

On my research, I came to know that as of now `enzyme` not providing an option to test for `:focus`.  
[Issue link](https://github.com/airbnb/enzyme/issues/703)

My unit test :point_down: 
```js
it( 'should not have autofocus if autofocus is false', () => {
	const wrapper = mount( <UrlInputButton autofocus={ false } /> );
	wrapper.setState( { expanded: true } );
	const elem = wrapper.find( 'input[type="text"]' );
	expect( elem.is( ':focus' ) ).toBe( false );
} );
```

**Comment by danielbachhuber** [comment]
@yahilmadakiya Looks fine to me. Can you go ahead and commit the unit test?

File: blocks/url-input/index.js [hunk 1]
```diff
@@ -180,11 +180,12 @@ class UrlInput extends Component {
 	render() {
 		const { value = '', instanceId } = this.props;
 		const { showSuggestions, posts, selectedSuggestion, loading } = this.state;
+		const autoFocus = 'undefined' === typeof this.props.autoFocus || this.props.autoFocus;
```
Comment (aduth) [inline blocks/url-input/index.js]: This is a good use case for destructuring default assignment, just as we're doing with `value`:

```js
const { value = '', autoFocus = true, instanceId } = this.props;
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Default_values_2

Comment (yahilmadakiya) [reply blocks/url-input/index.js]: @aduth Thanks.

**Comment by aduth** [comment]
For your test, could this work?

```js
it( 'should not have autofocus if autofocus is false', () => {
	const wrapper = mount( <UrlInputButton autofocus={ false } /> );
	wrapper.setState( { expanded: true } );
	const elem = wrapper.find( 'input[type="text"]' ).getDOMNode();
	expect( document.activeElement ).toBe( elem );
} );
```

(I've not tried it, and not sure if JSDOM is capable of assigning `activeElement` by `autofocus`)

**Comment by yahilmadakiya** [comment]
@aduth  I have already tried with `activeElement`. It was not working as per expectation.

Also, I have tried with the above test which you have provided. The Unit test passed on both case `autofocus={ true }` and `autofocus={ false }`.

**Thought**: `activeElement` has **Working Draft** status. So, I dont think we should use that.
https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/activeElement

**Comment by aduth** [comment]
That MDN link is unfortunately a bit confusing. I think the working draft status is referring to shadow DOM specifically. It is part of the WHATWG standard:

https://html.spec.whatwg.org/multipage/interaction.html#dom-document-activeelement

>The Unit test passed in both case autofocus={ true } and autofocus={ false }.

Could it be that this needs to be corrected to the camel-case form `autoFocus` ?

**Review by danielbachhuber — APPROVED** [review]
Given the difficulty of testing this, I think it's fine to land without tests.

**Comment by danielbachhuber** [comment]
Rebased with `master` and force-pushed to resolve merge conflict.

**Pull request merged**